src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJModelFactory.java

```diff
@@ -79,4 +88,48 @@ public Library createLibrary(@NotNull String name) {
     throw new UnsupportedOperationException(
         "Only common libraries like Scala SDK are currently supported.");
   }
+
+  /**
+   * This buddy here does all the nasty heavy lifting in order to check whether there are {@link
+   * Module} that depend on the removed {@link com.intellij.openapi.module.Module} and if finds,
+   * marks their {@link Module}s {@link StateMonitor} state.
+   *
+   * @param course            a {@link Course} which {@link Module}s to check
+   * @param removedModuleName a {@link String} name of the removed
+   * {@link com.intellij.openapi.module.Module}
+   */
+  public void markDependentModulesInvalid(@NotNull IntelliJCourse course,
+      @NotNull String removedModuleName) {
+    APlusProject project = course.getProject();
+    com.intellij.openapi.module.Module[] projectModules = project.getModuleManager().getModules();
```

Comment:
Could (and should) this use our domain model classes instead IDEA platform API?

Reply:
but it does, isn't it?
`project`**`.getModuleManager()`** is a method of the APlusProject class ("our API") or what do you mean? :confused:

Reply:
Yeah, I meant it could ask `Course` of the modules that are known to it (that is, defined in the course config file). I don't know if that would be much better solution. It could be implemented in `Module` class though as it wouldn't be IDEA-specific.

Reply:
I'm a bit confused, as there are currently two different entities (thus, under the same name):
* `com.intellij.openapi.module.Module` and 
* `fi.aalto.cs.apluscourses.intellij.model.IntelliJModule`

ideally, those two gentlemen would know about each other somehow :cupid: